server/repository/userProvider/update.ts

```diff
@@ -0,0 +1,24 @@
+import { BaseProtocol, DB } from '@server/db'
+import { Objects } from '@core/utils'
+import { User, UserAuthProvider } from '@meta/user'
+
+export const update = async (
+  props: {
+    user: Pick<User, 'id'>
+    password: string
+  },
+  client: BaseProtocol = DB
+): Promise<UserAuthProvider<{ password: string }>> => {
```

Comment:
I wonder if the return type is correct?
Shouldn't `returning *` return the insterted row from `public.users_auth_provider`?